app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -95,9 +99,29 @@ class StateFragmentPresenter @Inject constructor(
   private fun subscribeToCurrentState() {
     ephemeralStateLiveData.observe(fragment, Observer<EphemeralState> { result ->
```

Comment:
I don't think this is how we want the mechanism to work. Based on the current code, we're going to try to readd the content list over and over for each state change. Although the null check prevents that from happening, we won't properly rebind the content to the content list fragment.

Instead, we should be listening for the ephemeral state within ContentListFragment and depend on it there. That should simplify the arguments that need to be arranged here, and it keeps the fragments independent of one another.

Reply:
ephemeral state is added within ContentListFragment.